vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/InvalidCachedStatementExecutionEvent.java

```diff
@@ -0,0 +1,16 @@
+package io.vertx.sqlclient.impl;
+
+/**
+ * An event signals when an invalid cached prepared statement is executed and gets an error response.
+ */
+public class InvalidCachedStatementExecutionEvent {
+  private final PreparedStatement preparedStatement;
+
+  public InvalidCachedStatementExecutionEvent(PreparedStatement preparedStatement) {
```

Comment:
For now can we move this to the `io.vertx.sqlclient.impl.codec` package and rename to `InvalidCachedStatementEvent` because such detection could happen without the execution of a statement. Instead of the `PreparedStatement` just provide the statement identifier so it can be removed from the cache.